apps/website/src/components/chart/components/chart-component.tsx

```diff
@@ -0,0 +1,231 @@
+import { useMemo } from 'react'
+
+import { animated } from '@react-spring/web'
+import { Axis } from '@visx/axis'
+import { GridColumns } from '@visx/grid'
+import { Group } from '@visx/group'
+import { scaleLinear, scaleTime } from '@visx/scale'
+import { extent, max } from 'd3-array'
+
+import { getClosedIssues, getTotalIssues } from '../helpers/get-data'
+import { useAnimations, useChartTooltip } from '../hooks/'
+import { format } from '../utils/format-time'
+import { Areas } from './areas'
+import { ChartTooltip } from './chart-tooltip'
+import { Lines } from './lines'
+import { Markers } from './markers'
+
+import type { ChartProps } from '../chart'
+
+// defining colors
+const colors = {
+  total: '#E95460',
+  closed: '#23ADA0',
+  background: '#F0F2F5',
+  marker: '#09101C',
+  totalGradient: 'rgba(233, 84, 96, 0.3)',
+  closedGradient: 'rgba(35, 173, 160, 0.3)',
+  white: '#FFF',
+}
+
+const AnimatedGridColumns = animated(GridColumns)
+
+const ChartComponent = (props: ChartProps): JSX.Element => {
+  const { data, width: defaultWidth, height: defaultHeight } = props
+
+  // Extract dates, open issues, and closed issues from data
+  const dates = data.map(d => new Date(d.date))
+
+  // Define dimensions
+  const height = defaultHeight || 300
+  const width = defaultWidth || 300
+  const margin = { top: 20, right: 0, bottom: 30, left: 26 }
+  const innerWidth = (width || 0) - margin.left - margin.right
+  const innerHeight = height - margin.top - margin.bottom
+
+  const filteredDates = dates.filter(Boolean) // filters out undefined values
+  // Calculate total issues by summing open and closed issues
+  const totalIssues = data.map(d => d.open_issues + d.closed_issues)
+
+  const xDomain = extent(filteredDates) as [Date, Date]
+  const xScale = scaleTime({
+    domain: xDomain,
+    range: [0, innerWidth],
+  })
+
+  const numTicksBottom = Math.min(data.length, 12)
+
+  const yScale = scaleLinear({
+    domain: [0, max(totalIssues) || 0],
+    range: [innerHeight, 0],
+    nice: true,
+  })
+
+  const { tooltipData, handleTooltip, hideTooltip } = useChartTooltip({
```

Comment:
There is nothing wrong with this code, but whenever I see single word repeated many times, I think whether the hook API should be more like 
```ts
const tooltip = useChartTooltip({ ... })

console.log(tooltip.data)
console.log(tooltip.update({...}))
console.log(tooltip.hide())

```

Reply:
I am okay with both, just an observation.